src/goose/cli/session.py

```diff
@@ -158,20 +160,19 @@ def run(self) -> None:
             except Exception:
                 # rewind to right before the last user message
                 self.exchange.rewind()
-                print(traceback.format_exc())
+                print(traceback.lformat_exc())
                 print(
                     "\n[red]The error above was an exception we were not able to handle.\n\n[/]"
                     + "These errors are often related to connection or authentication\n"
                     + "We've removed the conversation up to the most recent user message"
                     + " - [yellow]depending on the error you may be able to continue[/]"
                 )
             self.notifier.stop()
-
+            save_latest_session(self.session_file_path, self.exchange.messages)
             print()  # Print a newline for separation.
             user_input = self.prompt_session.get_user_input()
             message = Message.user(text=user_input.text) if user_input.to_continue() else None
 
-        self.save_session()
```

Comment:
im not totally sure this is better, but if the session has been saving as current_session, this is maybe a good time to change it's name using the old dialog? otherwise i think we'll be constantly overwriting sessions

Reply:
just saw the existing discussion in #89 - even better to generate a name at the start and save to that (but avoid current session which will cause overwrites)